lib/shim/transaction-shim.js

```diff
@@ -254,33 +254,41 @@ function setTransactionName(name) {
  *  The transport type that brought the headers. Usually `HTTP` or `HTTPS`.
  */
 function handleCATHeaders(headers, segment, transportType) {
-  // Is CAT enabled?
-  if (!this.agent.config.cross_application_tracer.enabled) {
-    this.logger.trace('CAT disabled, not extracting header.')
+  // TODO: rename function or replace functionality when CAT fully removed.
+
+  if (!headers) {
+    this.logger.debug('No headers for CAT or DT processing.')
     return
   }
-  if (!this.agent.config.encoding_key) {
-    this.logger.warn('Missing encoding key, not extracting CAT headers!')
-    return
-  } else if (!headers) {
-    this.logger.debug('No headers to search for CAT within')
+
+  const config = this.agent.config
+
+  if (!config.cross_application_tracer.enabled && !config.distributed_tracing.enabled) {
```

Comment:
Short circuiting if both aren't enabled to avoid a small amount of work and reduce some nesting which should aid in the future code rip out. This can be reduced and the later DT check removed when we rip out CAT.